classes/04class/exercises/c04-iac04/raghunadhpokkalath/terraform-code/iac-04-module/variables.tf

```diff
@@ -0,0 +1,32 @@
+variable "vpc" {
+  type = map
+}
+
+variable "public_subnet_ids" {
+  type = list
+}
+
+variable "ami_id" {
+  type = string
+}
+
+variable "tags" {
+  type = map
+}
+
+variable "scales" {
+  type = map
+}
+
+variable user_data {
+  type = string
+  description = "User Data for EC2"
+  default = <<EOF
+    #! /bin/bash
+    yum update -y
+    yum install -y httpd
+    curl 169.254.169.254/latest/meta-data/hostname > index.html
+    mv index.html /var/www/html/
+    service httpd start
+  EOF
```

Comment:
User data usually is not a variable and the main reason for that is that it affects the bootstrap in EC2 instances. For this reason, the best practice is to version it in a VCS like git. 

If it is a variable any value could be passed in during runtime and the content would not be versioned.

Reply:
moved this to a sh file